src/Raven.Server/Config/JsonConfigFileModifier.cs

```diff
@@ -0,0 +1,226 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Raven.Client.Exceptions;
+using Raven.Client.Exceptions.Security;
+using Raven.Client.Json.Serialization.NewtonsoftJson.Internal;
+using Raven.Server.ServerWide;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Platform;
+using Sparrow.Server.Platform.Posix;
+using Sparrow.Utils;
+
+namespace Raven.Server.Config;
+
+public class JsonConfigFileModifier : IDisposable
+{
+    private readonly JsonOperationContext _context;
+    private readonly string _path;
+    private readonly bool _reset;
+    private BlittableJsonReaderObject _json;
+
+    public DynamicJsonValue DynamicJsonValue => _json.Modifications;
+
+    public static JsonConfigFileModifier Create(JsonOperationContext context, string path, bool reset = false)
+    {
+        var obj = new JsonConfigFileModifier(context, path, reset);
+        obj.Initial();
+        return obj;
+    }
+
+    protected JsonConfigFileModifier(JsonOperationContext context, string path, bool reset = false)
+    {
+        _context = context;
+        _path = path;
+        _reset = reset;
+    }
+
+    protected void Initial()
+    {
+        _json = ReadBlittableFromFile(_context);
+        _json.Modifications = new DynamicJsonValue(_json);
+    }
+    
+    public async Task Execute()
+    {
+        var modifiedJsonObj = _context.ReadObject(_json, "modified-settings-json");
+        await PersistConfiguration(modifiedJsonObj);
+    }
+
+    protected virtual void Validate(string path){}
+    
+    private BlittableJsonReaderObject ReadBlittableFromFile(JsonOperationContext context)
+    {
+        var jsonConfig = new DynamicJsonValue();
+        if (_reset == false)
+            ReadConfigFile(jsonConfig);
+
+        var fileName = Path.GetFileName(_path);
+        return context.ReadObject(jsonConfig, fileName);
+    }
+
+    private void ReadConfigFile(DynamicJsonValue jsonConfig)
+    {
+        try
+        {
+            using var fs = OpenFile(_path, FileMode.Open, FileAccess.ReadWrite);
```

Comment:
Why do you need `ReadWrite` permission here?

Reply:
The main issue is if I got permission exception from `File.Replace` I don't know if it is because of the main file or because of the backup file.
So I make sure I have `write` permission in this step since if I don't anyway there is nothing to do here.